package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -334,21 +355,34 @@ def _parse_atoms(self, datalines, massdict=None):
         n = len(datalines[0].split())
         has_charge = True if n in [7, 10] else False
 
-        types = np.zeros(n_atoms, dtype='|S5')
+        # atom ids aren't necessarily sequential
+        atom_ids = np.zeros(n_atoms, dtype=np.int32)
+        types = np.zeros(n_atoms, dtype=object)
         resids = np.zeros(n_atoms, dtype=np.int32)
         if has_charge:
             charges = np.zeros(n_atoms, dtype=np.float32)
 
-        for line in datalines:
+        for i, line in enumerate(datalines):
             line = line.split()
-            idx, resid = np.int64(line[:2])
-            atype = line[2]
-            idx -= 1
-            resids[idx] = resid
-            types[idx] = atype
+
+            atom_ids[i] = line[0]
+            resids[i] = line[1]
+            types[i] = line[2]
             if has_charge:
-                charge = float(line[3])
-                charges[idx] = charge
+                charges[i] = line[3]
```

Comment:
Yeah, because we knew the number of atoms a priori, we can create a np.array with the correct dtype and then just throw strings at it